scripts/west_commands/sign.py

```diff
@@ -391,6 +391,20 @@ def edt_flash_params(flash):
 
 class RimageSigner(Signer):
 
+    @staticmethod
+    def edt_get_rimage_target(board):
+        if 'up_squared_adsp' in board:
+            return 'apl'
+        if 'intel_adsp_cavs18' in board:
+            return 'cnl'
+        if 'intel_adsp_cavs20' in board:
+            return 'icl'
+        if 'intel_adsp_cavs25' in board:
+            return 'tgl'
```

Comment:
Is this really based on the board, or is it based on the SoC?

If it's based on the SoC, this seems kind of hard-coded. You have DTS files already, can't you do this based on an SoC-wide definition in there by accessing the edt object like the imgtool signer does?

Reply:
we still need to translate the soc as we have in zephyr to what the script needs (-m option)

Reply:
So you could get the soc in the DTS and then map it to the right -m option is what I'm hearing, right?

That would work regardless of how you name your boards.

Again, not a blocking comment, just wondering.

Reply:
sounds good, will look into it later.